Interfaces/GCClassic/main.F90

```diff
@@ -1763,8 +1799,33 @@ PROGRAM GEOS_Chem
              CALL Debug_Msg( '### MAIN: a DO_RRTMG_RAD_TRANSFER' )
           ENDIF
 
-          If (Allocated(DT_3D)) Deallocate(DT_3D)
-          If (Allocated(HR_3D)) Deallocate(HR_3D)
+          ! Copy the adjustment back to DT_3D as calculated in the baseline calculation
+          IF (Input_Opt%RRTMG_SEFDH) THEN
+              DT_3D(:,:,:) = DT_3D_UPDATE(:,:,:)
+          END IF
+
+          ! Store temperature change THEN heating rate from RRTMG in diagnostics (crb, 14/02/24)
+          If (Input_Opt%RRTMG_FDH) Then
+             IF (State_Diag%Archive_DynHeating) THEN
+                State_Diag%DynHeating(:,:,:) = HR_3D(:,:,:)
+             ENDIF
+             ! NB: DT_3D is the temperature adjustment either after equilibration (pure FDH)
+             ! or at the start of the NEXT radiation time step (SEFDH)
+             IF (State_Diag%Archive_DTRad     ) THEN
+                State_Diag%DTRad(:,:,:)      = DT_3D(:,:,:)
+             ENDIF
+             IF (Input_Opt%RRTMG_SEFDH) THEN
+                State_Chm%TStrat_Adj(:,:,:) = DT_3D(:,:,:)
+             END IF
+          ENDIF
+
+          RC = 0
+          IF (Allocated(DT_3D)) Deallocate(DT_3D, STAT=RC)
```

Comment:
It would be more efficient to allocate all necessary arrays once in initialization and then to deallocate them at finalaization rather than on every timestep.